### PR TITLE
EXUI-2019: View or Edit button not working on listed hearings

### PR DESCRIPTION
**Pull request by Josh-HMCTS**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/EXUI-2019

### Change description ###

Add in error message on hearings view when 'searchJudicialUserByPersonalCodes' returns 404
Allow navigation to hearings view when 'searchJudicialUserByPersonalCodes' returns 404

### CODE REVIEW

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ import { HearingsService } from '../../../services/hearings.service';
 import { HearingViewEditSummaryComponent } from './hearing-view-edit-summary.component';
 import { Section } from '../../../../hearings/models/section';
 import { ScreenNavigationModel } from '../../../../hearings/models/screenNavigation.model';
+import * as fromHearingStore from '../../../../hearings/store';
+import { Store } from '@ngrx/store';
```
Comment (connorpgpmcelroy) [inline src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts:14]: external imports should be above internal imports

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { HEARING_VIEW_EDIT_SUMMARY_TEMPLATE } from '../../../templates/hearing-v
 import { RequestHearingPageFlow } from '../request-hearing.page.flow';
 import { Section } from '../../../../hearings/models/section';
 import { ScreenNavigationModel } from '../../../../hearings/models/screenNavigation.model';
+import { HearingResponseError } from 'src/hearings/models/hearingResponseError.model';
```
Comment (connorpgpmcelroy) [inline src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts:15]: need to remove reference to src, should be ../../../../

File: src/hearings/resolvers/judicial-user-search-resolver.resolve.spec.ts [hunk 3]
```diff
@@ -3,15 +3,18 @@ import { HttpClientTestingModule } from '@angular/common/http/testing';
...
 import { RouterTestingModule } from '@angular/router/testing';
 import { provideMockStore } from '@ngrx/store/testing';
-import { of } from 'rxjs';
+import { of, throwError } from 'rxjs';
 import { initialStateImmutable } from '../hearing.test.data';
 import { JudicialUserModel } from '../models/judicialUser.model';
 import { JudicialRefDataService } from '../services/judicial-ref-data.service';
 import { JudicialUserSearchResolver } from './judicial-user-search-resolver.resolve';
+import * as fromHearingStore from '../store';
+import { Store } from '@ngrx/store';
```
Comment (connorpgpmcelroy) [inline src/hearings/resolvers/judicial-user-search-resolver.resolve.spec.ts:12]: Similar to previous

File: src/hearings/resolvers/judicial-user-search-response-resolver.resolve.spec.ts [hunk 4]
```diff
@@ -3,11 +3,13 @@ import { HttpClientTestingModule } from '@angular/common/http/testing';
...
 import { RouterTestingModule } from '@angular/router/testing';
 import { provideMockStore } from '@ngrx/store/testing';
-import { of } from 'rxjs';
+import { of, throwError } from 'rxjs';
 import { initialState } from '../hearing.test.data';
 import { JudicialUserModel } from '../models/judicialUser.model';
 import { JudicialRefDataService } from '../services/judicial-ref-data.service';
 import { JudicialUserSearchResponseResolver } from './judicial-user-search-response-resolver.resolve';
+import { Store } from '@ngrx/store';
+import * as fromHearingStore from '../store';
```
Comment (connorpgpmcelroy) [inline src/hearings/resolvers/judicial-user-search-response-resolver.resolve.spec.ts:12]: Similar to previous

File: src/hearings/store/actions/hearing-request.action.spec.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { HearingResponseError } from 'src/hearings/models/hearingResponseError.model';
```
Comment (connorpgpmcelroy) [inline src/hearings/store/actions/hearing-request.action.spec.ts:1]: similar to previous

**Review by connorpgpmcelroy — DISMISSED** [review]
Looks alright to be apart from minor import issues I can see

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts (hunk 1 above)
Comment (Josh-HMCTS) [reply src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts]: Moved

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts (hunk 2 above)
Comment (Josh-HMCTS) [reply src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.ts]: Removed ref to src

File: src/hearings/resolvers/judicial-user-search-resolver.resolve.spec.ts (hunk 3 above)
Comment (Josh-HMCTS) [reply src/hearings/resolvers/judicial-user-search-resolver.resolve.spec.ts]: Moved import

File: src/hearings/resolvers/judicial-user-search-response-resolver.resolve.spec.ts (hunk 4 above)
Comment (Josh-HMCTS) [reply src/hearings/resolvers/judicial-user-search-response-resolver.resolve.spec.ts:12]: Moved import

File: src/hearings/store/actions/hearing-request.action.spec.ts (hunk 5 above)
Comment (Josh-HMCTS) [reply src/hearings/store/actions/hearing-request.action.spec.ts]: Removed ref to src

**Review by andywilkinshmcts — APPROVED** [review]
Very nice fix

**Review by RiteshHMCTS — APPROVED** [review]

**Pull request merged**